### PR TITLE
Initial commit of Pediactric cptac publication study

### PR DESCRIPTION
**Pull request by migbro**
# What?
Fix # Load a published study of various pedicatric brain cancers at the request if Yichao Sun

Cancer studies updated in this pull request:
- a
- .

# checks
For all pull requests:
- [ ] Passes validation

For a new study (in addition to above):
- [ ] Does study name and study ID follow our convention? e.g. Tumor_Type (Institue, Journal Year); brca_mskcc_2015
- [ ] is study meta data complete? e.g. pmid, group of PUBLIC
- [x] were all samples profiled with WES/WGS? If not, is gene panel file curated?
- [ ] are oncotree codes of all samples curated; Cancer Type and Cancer Type Detailed needs to be added in addition to Oncotree Code
- [x] clinical sample and patient data with meta files
- [x] mutations data with meta files
- [ ] MAF is based on hg19
- [ ] MAF with 2 isoforms: uniprot and mskcc
- [x] CNA data with meta files
- [ ] CNA segment data with meta files
- [x] Expression data including z-scores with meta files
- [x] Case-lists for all profiles.
- [ ] Manual checking (Niki or JJ): Triage or private Portal link here

### CODE REVIEW

**Comment by yichaoS** [comment]
@migbro The PR is very well-done and we really appreciate the effort. There are small things related to format/naming here and there that needs to tweaked. 

I listed the points at the bottom (it's a bit overwhelming as we are going into details), along with some questions. I can also do the fixes for you tho, which would be quicker since we have the tools already set up and it could be quite tedious for you to investigate some of the points.  

We also need to set up a private testing instance to make sure the visualization is alright before pushing it to the public. I can help set up that as well and email your team to review. 

Clinical file: 
- different oncotree code (along with cancer type, cancer type detailed) should be added to clinical file as well 
- gender -> sex
- DFS months needs to be convereted to days so the survival curve could generate 

MAF
- 5 variants in MAF are without proper protein change annotation (see screenshot)
Screen Shot 2021-07-12 at 2 26 44 PM
- Was the MAF annotated by genome nexus? `NCBI_Build` column needs to be specified in MAF as `GRCh38`
- another MAF with msk annotation to be included

Other files:
- I saw in the paper the mRNA exp is done with RSEM, therefore rna seq files should be renamed to `data_mrna_seq_v2_rsem.txt` and the zscore file should be named as `data_mrna_seq_v2_rsem_zscores_ref_all_samples.txt` (if the referencing cohort is all samples) and generated with this script here: https://github.com/cBioPortal/datahub-study-curation-tools/tree/master/zscores/zscores_relative_allsamples
- file to be renamed: phosphosite -> phosphoprotein
- protein quantification case list should be named as `cases_protein_quantification.txt`. We have a tool to generate case lists here: https://github.com/cBioPortal/datahub-study-curation-tools/tree/master/case-list/generate_case_list

**Comment by migbro** [comment]
Hi @yichaoS , thanks for the review comments! I think I can answer some questions.
 - For oncotree code, do you mean to add that in as a separate field or to populate under `cancer_type_detailed`? Regardless I'd probably need some help assigning those. On our Kids First version of the portal, we have special designations for `High-grade glioma/astrocytoma` (`phgg`) and `Low-grade glioma/astrocytoma` (`plgg`) as they are different the adult versions. 
 - Are the docs alo out of date about gender/sex: `GENDER or SEX: Gender or sex of the patient (string)`? I could probably stand to capitalize  those standard fields so that the plots match the table view...
 - I am confused by this comment "DFS months needs to be convereted to days so the survival curve could generate..." the name of the field strongly suggests that the values be in months, not days...or at least the docs here: https://docs.cbioportal.org/5.1-data-loading/data-loading/file-formats#clinical-patient-columns should be amended to say that the column will be called `DFS_MONTHS`, but days are expected!
 - The MAFs were actually created using the MSKCC vcf2maf tool, v1.6.17. One difference was that VEP annotation was done using cache v93. Also, the NCBI build was dropped due to earlier version of cBioportal not accepting GRCh38 as an acceptable value. I can update this.
 - There might be a bug in the z scores calculated that I missed as I calculated them by column instead of row. That was unintended. I can and should fix that so that our version is also correct!

In short, I can apply some of the changes, I think the biggest help I might ask for would be in assigning oncotree codes. perhaps after I make some of the suggested updates we can go from there?

**Comment by yichaoS** [comment]
@migbro

- The `cancer type` and `cancer type details` are generated from oncotree code. But I looked into each cancer type you guys assigned and we currently don't have oncotree available for all of them. lemme discuss with @ritikakundra and get back to you on that one. 
- The docs are actually out of date for that part actually; we normalized all gender/sex into sex, so when user merge studies, the same clinical attributes could be merged successfully. 
- sorry about the DFS_MONTHS part, I was mixing it up with timeline but when I want to go back and fix that comment but you already replied. So please ignore that point. 
- For the MAF the coordinates are based on GRCh38 right? I'll need to check with our backend team for how we handling GRCh38 now and I'll get back to you on that for details.

**Comment by migbro** [comment]
@yichaoS this is helpful, thanks! You say that `cancer type` and `cancer type detailed` are generated from oncotree code...how so? If there is some automated way to convert our diagnoses to oncotree version, I am quite interested! The MAF coords are GRCh38. As far as I know, from working with @kalletlak this is "legal" now with some sort of patch database update, so I would assume that either some kind of liftover mechanism exists, or at the very least, inside that study all mutation function work just fine. If you are curious as to how the study looks (before updates you have suggested), you can see it here: https://pedcbioportal.kidsfirstdrc.org/study/summary?id=brain_cptac_2020

**Comment by yichaoS** [comment]
@migbro 

We have a simple tool here for generate `cancer type` and `cancer type details` from the oncotree code (https://github.com/cBioPortal/datahub-study-curation-tools/tree/master/clinical-data/add_CT_CTD) - not the other way around tho; the oncotree codes can to be assigned manually in the clinical file by looking up here: http://oncotree.mskcc.org/#/home

However, for `High-grade glioma/astrocytoma (phgg)` and `Low-grade glioma/astrocytoma (plgg)` we don't have those notes in current oncotree yet and @ritikakundra said we might want to add them. I'm replying to the email chain for this matter, please suggest where these two notes could potentially located there. 

And for `Craniopharyngioma` we have two notes related in oncotree, one is `Craniopharyngioma, Adamantinomatous Type` and one is `Craniopharyngioma, Adamantinomatous Type`, but I couldn't specify which should be used by looking through the data files in the PR. Could you let me know which one should we use? 

All other cancer types (except the above 3) provided in this PR can be mapped and I can help you adding that in the clinical files if you prefer, or simply provide you the type <-> code mapping and you could use our tool to generate the additional columns.

**Comment by migbro** [comment]
Hi @yichaoS , the `Craniopharyngioma` should be `acpg - Craniopharyngioma, Adamantinomatous Type`. I do have a table, that is attached that I have used in the past, but I am not sure if it still covers everything. 
[dx_index.txt](https://github.com/cBioPortal/datahub/files/6811892/dx_index.txt)

**Comment by yichaoS** [comment]
@migbro The mapping looks correct to me and I've follow up with an email about adding two oncotree notes.

**Comment by migbro** [comment]
I am working on this today, so I am commenting to keep track of changes I will make and commit:
 - [ ] Add onocotree code as a column in clinical file
 - [ ] Populate `CANCER_TYPE_DETAILED` with values of oncotree code, i.e.
 
        |CANCER_TYPE_DETAILED            |ONCOTREE_CODE|
        |--------------------------------|-------------|
        |Atypical Teratoid/Rhabdoid Tumor|ATRT         |

 - [ ] Change `GENDER` to `SEX`
 - [ ] Rename RNA data files to `data_mrna_seq_v2_rsem.txt` and `data_mrna_seq_v2_rsem_zscores_ref_all_samples.txt`
 - [ ] Rename `phosphosite -> phosphoprotein`
 - [ ] Create `cases_protein_quantification.txt` - will consider using tool mentioned here: https://github.com/cBioPortal/datahub-study-curation-tools/tree/master/case-list/generate_case_list to achieve

The only thing left I am uncertain on how to resolve are the 5 row entries in the maf file that do not have proper protein change info

**Comment by migbro** [comment]
Hi @yichaoS , please review my last comments in relation to the commits I have just made. I think I have addressed almost all of your concerns. Please let me know if it looks ok. I also added some organizational priority to the clinical files.

**Comment by migbro** [comment]
@yichaoS just a quick update on this, I loaded this onto our QA server ans noticed that the changes cause the sample yo not load. After troubleshooting, I realized the case of the oncotree code was throwing it off. So, I made it lower case, and now it displays just fine.

**Comment by yichaoS** [comment]
@migbro study looks good. I'm discussing with the group about our infrastructure for handling hg38 studies - will try to get the study imported once we've settled for that - I'll keep you posted. Thanks for all your work!

**Review by yichaoS — APPROVED** [review]

**Comment by migbro** [comment]
Yes, the doi is listed at the end of the description: `https://doi.org/10.1016/j.cell.2020.10.044`
I did try using the pmid feature, but that just seems tack on to the description which has a 1,024 character limit. Not sure how much more I could trim

**Comment by SRodenburg** [comment]
FYI; we used liftover before to translate hg38 coordinates to hg19.
